### PR TITLE
Google Client sheet create/update/delete functionalities

### PR DESCRIPTION
**Pull request by neerajadobe93**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   },
   "scripts": {
     "test": "c8 mocha",
+    "testd": "c8 mocha --inspect-brk",
```
Comment (tripodsan) [inline package.json:12]: do you really need this? usually this is automatically done via the IDE.

File: src/GoogleClient.d.ts [hunk 2]
```diff
@@ -156,4 +156,35 @@ declare class GoogleClient {
    * @returns {Promise<object>|null} The document or {@code null} if the document does not exist
    */
   getDocumentFromPath(parentId:string, path:string, noRetry:boolean):Promise<object>;
+
+   /**
+   *  create a new file in the given parent folder with the given name and mimetype
+   *  @param {string} parentId
+   * @param {string} name
+   * @param {string} mimeType either {@link GoogleClient.TYPE_DOCUMENT} or {@link GoogleClient.TYPE_SPREADSHEET}
+   * @returns {Promise<Object>} file object
```
Comment (tripodsan) [inline src/GoogleClient.d.ts:165]:
```suggestion
   * @returns {Promise<object>} file object
```

File: src/GoogleClient.d.ts [hunk 3]
```diff
@@ -156,4 +156,35 @@ declare class GoogleClient {
...
+   */
+   createBlankDocOrSheet(parentId:string, name:string, mimeType:string):Promise<Object>;
```
Comment (tripodsan) [inline src/GoogleClient.d.ts:167]:
```suggestion
   createBlankDocOrSheet(parentId:string, name:string, mimeType:string):Promise<object>;
```

File: src/GoogleClient.d.ts [hunk 4]
```diff
@@ -156,4 +156,35 @@ declare class GoogleClient {
...
+
+
+   /**
+    * 
+    * @param {string} spreadsheetId 
+    * @param {string} sheetName 
+    * @param {object} worksheetData 
+    * 
+    * @returns {Promise<Object>} GSheet object
```
Comment (tripodsan) [inline src/GoogleClient.d.ts:176]:
```suggestion
    * @returns {Promise<object>} GSheet object
```

File: src/GoogleClient.d.ts [hunk 5]
```diff
@@ -156,4 +156,35 @@ declare class GoogleClient {
...
+    */
+   createOrUpdateGSheet(spreadsheetId:string, sheetName:string, worksheetData:object):Promise<Object>;
```
Comment (tripodsan) [inline src/GoogleClient.d.ts:178]: I would pass the `create` flag, so it's also possible to fail if not exists.

```suggestion
   updateSheet(spreadsheetId:string, sheetName:string, worksheetData:object, create:boolean):Promise<object>;
```

**Review by tripodsan — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (neerajadobe93) [reply package.json:12]: sry for this , mistakely checked in

**Review by tripodsan — APPROVED** [review]

**Pull request merged**